Realm/RLMResults.mm

```diff
@@ -560,6 +592,15 @@ - (id)objectAtIndex:(NSUInteger)index {
     @throw RLMException(@"Index is out of bounds.", @{@"index": @(index)});
 }
 
+- (id)valueForKey:(NSString *)key {
+    RLMResultsValidate(self);
+    return @[];
+}
+
+- (void)setValue:(__unused id)value forKey:(__unused NSString *)key {
+    RLMResultsValidateInWriteTransaction(self);
```

Comment:
Good points, can you add validations elsewhere in this class? That's small enough to do in this PR.